### PR TITLE
fix not login and run a cloudImage will failed

### PR DESCRIPTION
**Pull request by bxy4543**
fix not login and run a cloudImage will failed

### CODE REVIEW

File: runtime/utils.go [hunk 1]
```diff
@@ -62,11 +63,14 @@ func PreInitMaster0(sshClient ssh.Interface, remoteHostIP string) error {
...
+	authFile := common.DefaultRegistryAuthConfigDir()
+	//if auth file exist, copy file to remote
+	if _, err := os.Stat(authFile); err == nil {
+		// send register login info
+		err = sshClient.Copy(remoteHostIP, authFile, authFile)
+		if err != nil {
+			return fmt.Errorf("send register config to remote host %s failed:%v", remoteHostIP, err)
+		}
 	}
 	logger.Info("send register info to %s success !", remoteHostIP)
```
Comment (justadogistaken) [inline runtime/utils.go:75]: what is the solution if authFile not exists? still log success?

Comment (justadogistaken) [reply runtime/utils.go:71]: send registry auth info , not register info.
and the success log would still be executed, I it's not allowed

File: runtime/utils.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func PreInitMaster0(sshClient ssh.Interface, remoteHostIP string) error {
 	// send register login info
 	err = sshClient.Copy(remoteHostIP, common.DefaultRegistryAuthConfigDir(), common.DefaultRegistryAuthConfigDir())
```
Comment (justadogistaken) [inline runtime/utils.go:67]: we'd better check file existence first

File: runtime/utils.go [hunk 3]
```diff
@@ -64,11 +64,14 @@ func PreInitMaster0(sshClient ssh.Interface, remoteHostIP string) error {
...
-	err = sshClient.Copy(remoteHostIP, common.DefaultRegistryAuthConfigDir(), common.DefaultRegistryAuthConfigDir())
-	if err != nil {
-		return fmt.Errorf("send register config to remote host %s failed:%v", remoteHostIP, err)
+	authFile := common.DefaultRegistryAuthConfigDir()
+	if utils.IsFileExist(authFile) {
+		err = sshClient.Copy(remoteHostIP, authFile, authFile)
+		if err != nil {
+			return fmt.Errorf("failed to send register config %s to remote host %s err: %v", authFile, remoteHostIP, err)
+		}
+		logger.Info("send register info to %s success !", remoteHostIP)
```
Comment (justadogistaken) [inline runtime/utils.go:73]: we should ensure whether authFile is necessary, and if we should warn users when the file does not exist

**Comment by fanux** [comment]
LGTM  @justadogistaken

**Review by fanux — APPROVED** [review]

**Review by justadogistaken — APPROVED** [review]

**Pull request merged**